--- a/NeoForge/build.gradle
+++ b/NeoForge/build.gradle
@@ -1,5 +1,5 @@
 // Adjust the output jar name here
-archivesBaseName = "${mod_name.replace(" ", "")}-NeoForge-${minecraft_version}"
+archivesBaseName = "${mod_name.replace(" ", "")}-Forge-${minecraft_version}"
 
 dependencies {
     // Add your dependencies here
@@ -15,7 +15,7 @@
  */
 
 unimined.minecraft {
-    neoForge {
+    minecraftForge {
         if (isPort) {
             loader "net.neoforged:neoforge:${neoforge_version}:universal"
         } else {
@@ -29,7 +29,7 @@
     from project(":Common").sourceSets.main.resources
     def buildProps = project.properties.clone()
 
-    filesMatching("META-INF/neoforge.mods.toml") {
+    filesMatching("META-INF/mods.toml") {
         expand buildProps
     }
 }
@@ -53,8 +53,8 @@
     setVersionType("release")
     setChangelog("https://raw.githubusercontent.com/hypherionmc/changelogs/refs/heads/main/pocketmachines.md")
     setProjectVersion("${minecraft_version}-${project.version}")
-    setDisplayName("[NeoForge 1.21.9] Simple Pocket Machines - ${project.version}")
-    setGameVersions("1.21.9")
+    setDisplayName("[Forge/NeoForge 1.20.1] Simple Pocket Machines - ${project.version}")
+    setGameVersions("1.20.1")
     setLoaders("neoforge")
     setArtifact(remapJar)
     setCurseEnvironment("both")
