--- a/Common/src/main/java/com/hypherionmc/pocketmachines/common/items/PocketEnchanter.java
+++ b/Common/src/main/java/com/hypherionmc/pocketmachines/common/items/PocketEnchanter.java
@@ -1,6 +1,7 @@
 package com.hypherionmc.pocketmachines.common.items;
 
 import com.hypherionmc.pocketmachines.common.items.base.BasePocketGuiItem;
+import lombok.NoArgsConstructor;
 import net.minecraft.core.BlockPos;
 import net.minecraft.network.chat.Component;
 import net.minecraft.world.InteractionHand;
@@ -12,13 +13,8 @@
 import net.minecraft.world.level.Level;
 import org.jetbrains.annotations.NotNull;
 
-import static com.hypherionmc.pocketmachines.ModConstants.makeResourceKey;
-
+@NoArgsConstructor
 public class PocketEnchanter extends BasePocketGuiItem {
-
-    public PocketEnchanter() {
-        super(makeResourceKey("pocket_enchanting_table"));
-    }
 
     @Override
     public void openMenu(@NotNull Level level, @NotNull Player player, @NotNull InteractionHand hand) {
