--- a/Common/src/main/java/com/hypherionmc/pocketmachines/common/inventory/PocketChestInventory.java
+++ b/Common/src/main/java/com/hypherionmc/pocketmachines/common/inventory/PocketChestInventory.java
@@ -16,8 +16,6 @@
 import net.minecraft.world.inventory.AbstractContainerMenu;
 import net.minecraft.world.inventory.ChestMenu;
 import net.minecraft.world.item.ItemStack;
-import net.minecraft.world.level.storage.ValueInput;
-import net.minecraft.world.level.storage.ValueOutput;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
@@ -29,15 +27,15 @@
     private static final Component TITLE = Component.translatable("item.pocketmachines.pocket_chest");
     protected final NonNullList<ItemStack> items = NonNullList.withSize(INVENTORY_SIZE, ItemStack.EMPTY);
 
-    public PocketChestInventory(ValueInput input) {
+    public PocketChestInventory(CompoundTag tag, HolderLookup.Provider provider) {
         super(INVENTORY_SIZE);
 
-        if (input != null)
-            this.load(input);
+        if (tag != null && provider != null)
+            this.load(tag, provider);
     }
 
     public PocketChestInventory() {
-        this(null);
+        this(null, null);
     }
 
     @Nullable
@@ -51,13 +49,13 @@
         return TITLE;
     }
 
-    public void load(ValueInput tag) {
+    public void load(CompoundTag tag, HolderLookup.Provider provider) {
         this.items.clear();
-        ContainerHelper.loadAllItems(tag, this.items);
+        ContainerHelper.loadAllItems(tag, this.items, provider);
     }
 
-    public void save(@NotNull ValueOutput tag) {
-        ContainerHelper.saveAllItems(tag, this.items);
+    public void save(@NotNull CompoundTag tag, HolderLookup.@NotNull Provider provider) {
+        ContainerHelper.saveAllItems(tag, this.items, provider);
     }
 
     @Override
