--- a/Common/src/main/java/com/hypherionmc/pocketmachines/common/inventory/PocketBlastFurnaceInventory.java
+++ b/Common/src/main/java/com/hypherionmc/pocketmachines/common/inventory/PocketBlastFurnaceInventory.java
@@ -8,13 +8,12 @@
 import net.minecraft.world.entity.player.Inventory;
 import net.minecraft.world.inventory.AbstractContainerMenu;
 import net.minecraft.world.item.crafting.RecipeType;
-import net.minecraft.world.level.storage.ValueInput;
 import org.jetbrains.annotations.NotNull;
 
 public class PocketBlastFurnaceInventory extends AbstractPocketFurnaceInventory {
 
-    public PocketBlastFurnaceInventory(ValueInput input) {
-        super(RecipeType.BLASTING, input, Component.translatable("item.pocketmachines.pocket_blast_furnace"));
+    public PocketBlastFurnaceInventory(CompoundTag tag, HolderLookup.Provider provider) {
+        super(RecipeType.BLASTING, tag, provider, Component.translatable("item.pocketmachines.pocket_blast_furnace"));
     }
 
     public PocketBlastFurnaceInventory() {
