--- a/Common/src/main/java/com/hypherionmc/pocketmachines/common/items/PocketBrewingStand.java
+++ b/Common/src/main/java/com/hypherionmc/pocketmachines/common/items/PocketBrewingStand.java
@@ -10,12 +10,10 @@
 import net.minecraft.world.level.Level;
 import org.jetbrains.annotations.NotNull;
 
-import static com.hypherionmc.pocketmachines.ModConstants.makeResourceKey;
-
 public class PocketBrewingStand extends BaseTickablePocketItem<PocketBrewingStandInventory> {
 
     public PocketBrewingStand() {
-        super(PersistedMachines.POCKET_BREWING_STAND, "TG_POCKET_BREWING_STAND_INDEX", makeResourceKey("pocket_brewing_stand"));
+        super(PersistedMachines.POCKET_BREWING_STAND, "TG_POCKET_BREWING_STAND_INDEX");
     }
 
     @Override
