--- a/Common/src/main/java/com/hypherionmc/pocketmachines/common/inventory/PocketFurnaceInventory.java
+++ b/Common/src/main/java/com/hypherionmc/pocketmachines/common/inventory/PocketFurnaceInventory.java
@@ -8,13 +8,12 @@
 import net.minecraft.world.entity.player.Inventory;
 import net.minecraft.world.inventory.AbstractContainerMenu;
 import net.minecraft.world.item.crafting.RecipeType;
-import net.minecraft.world.level.storage.ValueInput;
 import org.jetbrains.annotations.NotNull;
 
 public class PocketFurnaceInventory extends AbstractPocketFurnaceInventory {
 
-    public PocketFurnaceInventory(ValueInput input) {
-        super(RecipeType.SMELTING, input, Component.translatable("item.pocketmachines.pocket_furnace"));
+    public PocketFurnaceInventory(CompoundTag tag, HolderLookup.Provider provider) {
+        super(RecipeType.SMELTING, tag, provider, Component.translatable("item.pocketmachines.pocket_furnace"));
     }
 
     public PocketFurnaceInventory() {
