--- a/Fabric/src/main/java/com/hypherionmc/pocketmachines/PocketMachinesFabric.java
+++ b/Fabric/src/main/java/com/hypherionmc/pocketmachines/PocketMachinesFabric.java
@@ -1,9 +1,12 @@
 package com.hypherionmc.pocketmachines;
 
 import com.hypherionmc.pocketmachines.common.setup.CommonSetup;
+import com.hypherionmc.pocketmachines.common.setup.ModItems;
+import com.hypherionmc.pocketmachines.common.setup.ModTabs;
 import com.hypherionmc.pocketmachines.common.world.PersistedMachines;
 import net.fabricmc.api.ModInitializer;
 import net.fabricmc.fabric.api.event.lifecycle.v1.ServerWorldEvents;
+import net.fabricmc.fabric.api.itemgroup.v1.ItemGroupEvents;
 
 public class PocketMachinesFabric implements ModInitializer {
 
