--- a/Fabric/build.gradle
+++ b/Fabric/build.gradle
@@ -52,8 +52,8 @@
     setVersionType("release")
     setChangelog("https://raw.githubusercontent.com/hypherionmc/changelogs/refs/heads/main/pocketmachines.md")
     setProjectVersion("${minecraft_version}-${project.version}")
-    setDisplayName("[FABRIC/QUILT 1.21.11] Simple Pocket Machines - ${project.version}")
-    setGameVersions("1.21.11")
+    setDisplayName("[FABRIC/QUILT 1.21.4] Simple Pocket Machines - ${project.version}")
+    setGameVersions("1.21.4")
     setLoaders("fabric", "quilt")
     setArtifact(remapJar)
     setCurseEnvironment("both")
