--- a/Common/src/main/java/com/hypherionmc/pocketmachines/common/inventory/PocketBrewingStandInventory.java
+++ b/Common/src/main/java/com/hypherionmc/pocketmachines/common/inventory/PocketBrewingStandInventory.java
@@ -173,12 +173,12 @@
     public void load(CompoundTag compoundTag, HolderLookup.Provider provider) {
         this.items.clear();
         ContainerHelper.loadAllItems(compoundTag, this.items, provider);
-        this.brewTime = compoundTag.getShortOr("BrewTime", (short) 0);
+        this.brewTime = compoundTag.getShort("BrewTime");
         if (this.brewTime > 0) {
             this.ingredient = this.items.get(3).getItem();
         }
 
-        this.fuel = compoundTag.getByteOr("Fuel", (byte) 0);
+        this.fuel = compoundTag.getByte("Fuel");
     }
 
     public void save(@NotNull CompoundTag compoundTag, HolderLookup.@NotNull Provider provider) {
